### PR TITLE
fix(argo-cd): Always create ApplicationSet as following upstream

### PR DESCRIPTION
**Pull request by yu-croco**
Related to https://github.com/argoproj/argo-helm/pull/2682

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-cd/values.yaml [hunk 1]
```diff
@@ -2638,9 +2638,6 @@ repoServer:
 
 ## ApplicationSet controller
 applicationSet:
-  # -- Enable ApplicationSet controller
-  enabled: true
```
Comment (yu-croco) [inline charts/argo-cd/values.yaml:2642]: Is it better to write changelog in README...? 🤔

**Review by mbevc1 — APPROVED** [review]

**Review by pdrastil — APPROVED** [review]

File: charts/argo-cd/values.yaml (hunk 1 above)
Comment (pdrastil) [reply charts/argo-cd/values.yaml:2642]: I would mention that even in readme as enteprise users with proxy repositories might be suprised that this is now always installed.

Comment (yu-croco) [reply charts/argo-cd/values.yaml:2642]: @pdrastil 
Thank you for your comment. I will add 🙋

Comment (yu-croco) [reply charts/argo-cd/values.yaml:2642]: I added in this commit https://github.com/argoproj/argo-helm/pull/2688/commits/822ab42195754e79af05f4adb783295c438e2f61 . 🫡

**Review by mkilchhofer — APPROVED** [review]
Thanks @yu-croco 👍👌

**Comment by rouke-broersma** [comment]
Based on the original proposal for merging the appset codebase with the argocd codebase it seems to me like it was intended for appset to be optional. Therefore it seems like this is a bug in argocd and the correct fix would be to remove the requirement in argocd. See: https://github.com/argoproj/argo-cd/pull/8743#discussion_r824240111

**Comment by pdrastil** [comment]
Hi @hsmade I'm not sure why installation would be impossible. As you state you have centrally managed Argo by some team. Same team can also install AppSet controller as they would be cluster admins anyway that are allowed to install / upgrade CRDs. AppSet resource is restricted only to the namespace where Argo is installed and resources are usable only by admins via kubectl or GitOps (you cannot create AppSet resource via UI as end user). In multi-tenant setup it's a good practice to restrict Argo namespace only to admins and do not allow tenants modify it as it contains sensitive integration secrets for whole organisation. Tenants in such case are usually given their own project that allows them specific resources (not AppSet, CRDs or other global resources) / namespaces (not argo, kube-system, etc.) / OIDC binding (not using admin account) so they can self-provision their application. Use of Application in any namespace feature is also tied to correct use of projects if you want to give users access to K8s cluster via kubectl.

**Comment by hsmade** [comment]
>As you state you have centrally managed Argo by some team
No, we install argo on a centrally managed k8s cluster. So we don't own the k8s cluster ourselves, and there's other teams that also install argo on their namespaces. So the argo we install, only manages our own namespace(s). Therefor, the cluster admins limit what CRDs we can access.

**Comment by felipeng** [comment]
> Based on the original proposal for merging the appset codebase with the argocd codebase it seems to me like it was intended for appset to be optional. Therefore it seems like this is a bug in argocd and the correct fix would be to remove the requirement in argocd. See: [argoproj/argo-cd#8743 (comment)](https://github.com/argoproj/argo-cd/pull/8743#discussion_r824240111)
> 
> Wouldn't it make more sense to allow deploying the CRD without the applicationset controller? I don't use the appset controller and I don't want to use the appset controller so this is just wasting resources in my cluster for no benefit.

Same use case here. We don't use AppSet, and it is not possible to avoid its creation. I set AppSet replicas to 0; however, there are other resources created. Not a big deal, but why create Kubernetes objects if they will not be used?
https://github.com/argoproj/argo-helm/issues/4043

**Pull request merged**